crates/loader/src/local.rs

```diff
@@ -394,17 +397,40 @@ impl LocalLoader {
                     quoted_path(&dest_parent)
                 )
             })?;
-        crate::fs::copy(src, dest).await.with_context(|| {
-            format!(
-                "Failed to copy {} to {}",
-                quoted_path(src),
-                quoted_path(dest)
-            )
-        })?;
+        crate::fs::copy(src, dest)
+            .await
+            .or_else(|e| Self::failed_to_copy_single_file_error(src, dest, guest_dest, e))?;
         tracing::debug!("Copied {src:?} to {dest:?}");
         Ok(())
     }
 
+    fn failed_to_copy_single_file_error<T>(
+        src: &Path,
+        dest: &Path,
+        guest_dest: &str,
+        e: anyhow::Error,
+    ) -> anyhow::Result<T> {
+        let src_text = quoted_path(src);
+        let dest_text = quoted_path(dest);
+        let base_msg = format!("Failed to copy {src_text} to working path {dest_text}");
+
+        if let Some(io_error) = e.downcast_ref::<std::io::Error>() {
+            if Self::is_directory_like(guest_dest)
+                || io_error.kind() == std::io::ErrorKind::NotFound
+            {
+                bail!("{base_msg}: \"{guest_dest}\" is not a valid destination file name");
```

Comment:
nit: could we make everything after the `base_msg` into another error (perhaps by using `with_context`) so that that information shows up on a different line when displaying the full error. As it is right now, this error message is very long.

Reply:
Good call.  I broke the message up and it now looks like:

```
$ spin up -f ./crates/loader/tests/file-errors/bad.toml 
Error: Failed to load manifest from "./crates/loader/tests/file-errors/bad.toml"

Caused by:
   0: Failed to load Spin app from "./crates/loader/tests/file-errors/bad.toml"
   1: Failed to load component `bad`
   2: Failed to copy "/home/ivan/github/spin/crates/loader/tests/file-errors/host.txt" to working path "/tmp/spinup-iloZyr/assets/bad/"
   3: "/" is not a valid destination file name
```

which draws out the underlying error much more clearly.  Thanks!